### PR TITLE
svc: Add all missing SVC stubs

### PR DESCRIPTION
**Pull request by SciresM**

### CODE REVIEW

File: nx/include/switch/kernel/svc.h [hunk 1]
```diff
@@ -597,27 +633,65 @@ Result svcGetInfo(u64* out, u32 id0, Handle handle, u64 id1);
...
+///@name Process and thread management
+///@{
+
+/**
+ * @brief Gets information about a thread that will be scheduled in the future. [5.0.0+]
+ * @return Result code.
+ * @note Syscall number 0x2E.
+ * @warning This is a privileged syscall. Use \ref envIsSyscallHinted to check if it is available.
+ */
+Result GetDebugFutureThreadInfo(LastThreadContext *out_context, u64 *out_thread_id, Handle debug, s64 ns);
```
Comment (yellows8) [inline nx/include/switch/kernel/svc.h]: svc prefix missing. Doc all params (same elsewhere).

File: nx/include/switch/kernel/svc.h [hunk 2]
```diff
@@ -597,27 +633,65 @@ Result svcGetInfo(u64* out, u32 id0, Handle handle, u64 id1);
 
 ///@}
 
+///@name Cache Management
+///@{
+
+/**
+ * @brief Flushes the entire data cache (by set/way).
+ * @note Syscall number 0x2A.
+ * @warning This is a privileged syscall. Use \ref envIsSyscallHinted to check if it is available.
```
Comment (fincs) [inline nx/include/switch/kernel/svc.h]: @TuxSH Should a note be added saying that this syscall is dangerous and shouldn't be used?

File: nx/include/switch/kernel/svc.h [hunk 3]
```diff
@@ -597,27 +633,65 @@ Result svcGetInfo(u64* out, u32 id0, Handle handle, u64 id1);
...
+ */
+Result GetDebugFutureThreadInfo(LastThreadContext *out_context, u64 *out_thread_id, Handle debug, s64 ns);
+
+/**
+ * @brief Gets information about the previously-scheduled thread.
+ * @return Result code.
+ * @note Syscall number 0x2F.
+ */
+Result GetLastThreadInfo(LastThreadContext *out_context, u64 *out_tls_address, u32 *out_flags);
```
Comment (fincs) [inline nx/include/switch/kernel/svc.h]: Missing svc prefix - also, is this really a non-privileged syscall?

File: nx/include/switch/kernel/svc.h [hunk 4]
```diff
@@ -661,6 +735,76 @@ Result svcGetThreadContext3(ThreadContext* ctx, Handle thread);
...
+///@name Debugging
+///@{
+/**
+ * @brief Causes the kernel to dump debug information. [1.0.0-3.0.2]
+ * @param[in] dump_info_type Type of information to dump.
+ * @param[in] arg0 Argument to the debugging operation.
+ * @note Syscall number 0x3C.
+ * @warning This is a privileged syscall. Use \ref envIsSyscallHinted to check if it is available.
+ */
+void DumpInfo(u32 dump_info_type, u64 arg0);
```
Comment (fincs) [inline nx/include/switch/kernel/svc.h]: Missing svc prefix

File: nx/include/switch/kernel/svc.h [hunk 5]
```diff
@@ -661,6 +735,76 @@ Result svcGetThreadContext3(ThreadContext* ctx, Handle thread);
...
+/**
+ * @brief Performs a debugging operation on the kernel. [4.0.0+]
+ * @param[in] kern_debug_type Type of debugging operation to perform.
+ * @param[in] arg0 First argument to the debugging operation.
+ * @param[in] arg1 Second argument to the debugging operation.
+ * @param[in] arg2 Third argument to the debugging operation.
+ * @note Syscall number 0x3C.
+ * @warning This is a privileged syscall. Use \ref envIsSyscallHinted to check if it is available.
+ */
+void KernelDebug(u32 kern_debug_type, u64 arg0, u64 arg1, u64 arg2);
```
Comment (fincs) [inline nx/include/switch/kernel/svc.h]: Missing svc prefix

File: nx/include/switch/kernel/svc.h [hunk 6]
```diff
@@ -661,6 +735,76 @@ Result svcGetThreadContext3(ThreadContext* ctx, Handle thread);
...
+
+/**
+ * @brief Performs a debugging operation on the kernel. [4.0.0+]
+ * @param[in] kern_trace_state Type of tracing the kernel should perform.
+ * @note Syscall number 0x3D.
+ * @warning This is a privileged syscall. Use \ref envIsSyscallHinted to check if it is available.
+ */
+void ChangeKernelTraceState(u32 kern_trace_state);
```
Comment (fincs) [inline nx/include/switch/kernel/svc.h]: Missing svc prefix

File: nx/include/switch/kernel/svc.h [hunk 7]
```diff
@@ -888,6 +1075,38 @@ Result svcUnmapDeviceAddressSpace(Handle handle, Handle proc_handle, u64 map_add
...
+///@{
+
+/**
+ * @brief Invalidates data cache for a virtual address range within a process.
+ * @param[in] address Address of region to invalidate.
+ * @param[in] size Size of region to invalidate.
+ * @note Syscall number 0x5D.
+ * @warning This is a privileged syscall. Use \ref envIsSyscallHinted to check if it is available.
+ */
+Result svcInvalidateProcessDataCache(Handle process, uintptr_t address, size_t size);
```
Comment (fincs) [inline nx/include/switch/kernel/svc.h]: Should use void* for address

File: nx/include/switch/kernel/svc.h [hunk 8]
```diff
@@ -888,6 +1075,38 @@ Result svcUnmapDeviceAddressSpace(Handle handle, Handle proc_handle, u64 map_add
...
+
+/**
+ * @brief Stores data cache for a virtual address range within a process.
+ * @param[in] address Address of region to store.
+ * @param[in] size Size of region to store.
+ * @note Syscall number 0x5E.
+ * @warning This is a privileged syscall. Use \ref envIsSyscallHinted to check if it is available.
+ */
+Result svcStoreProcessDataCache(Handle process, uintptr_t address, size_t size);
```
Comment (fincs) [inline nx/include/switch/kernel/svc.h]: Should use void* for address

File: nx/include/switch/kernel/svc.h [hunk 9]
```diff
@@ -888,6 +1075,38 @@ Result svcUnmapDeviceAddressSpace(Handle handle, Handle proc_handle, u64 map_add
...
+
+/**
+ * @brief Flushes data cache for a virtual address range within a process.
+ * @param[in] address Address of region to flush.
+ * @param[in] size Size of region to flush.
+ * @note Syscall number 0x5F.
+ * @warning This is a privileged syscall. Use \ref envIsSyscallHinted to check if it is available.
+ */
+Result svcFlushProcessDataCache(Handle process, uintptr_t address, size_t size);
```
Comment (fincs) [inline nx/include/switch/kernel/svc.h]: Should use void* for address

File: nx/include/switch/kernel/svc.h [hunk 10]
```diff
@@ -597,27 +633,65 @@ Result svcGetInfo(u64* out, u32 id0, Handle handle, u64 id1);
...
+ */
+void svcFlushEntireDataCache(void);
+
+/**
+ * @brief Flushes data cache for a virtual address range.
+ * @param[in] address Address of region to flush.
+ * @param[in] size Size of region to flush.
+ * @note Syscall number 0x2B.
+ */
+Result svcFlushDataCache(void *address, size_t size);
```
Comment (fincs) [inline nx/include/switch/kernel/svc.h]: Add note saying that armDCacheFlush should be used instead

File: nx/include/switch/kernel/svc.h (hunk 3 above)
Comment (SciresM) [reply nx/include/switch/kernel/svc.h]: it is not privileged.

**Pull request merged**